dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyLister.java

```diff
@@ -0,0 +1,135 @@
+ * (formerly known as Eclipse Aether).
+ */
+public class DependencyLister {
+  
+  private static final RepositorySystem SYSTEM = newRepositorySystem();
+  private static final RemoteRepository CENTRAL =
+      new RemoteRepository.Builder("central", "default", "http://repo1.maven.org/maven2/").build();
+
+  public static void main(String[] args)
+      throws DependencyCollectionException, DependencyResolutionException {
+    
+    if (args.length != 1 || !args[0].contains(":")) {
+      System.err.println("Usage: java "
+          + "com.google.cloud.tools.opensource.dependencies.Dependencies "
+          + "groupdId:artifactId:version");
+      return;
+    }
+    
+    String[] coordinates = args[0].split(":");
+    String groupId = coordinates[0];
+    String artifactId = coordinates[1];
+    String version = coordinates[2];
+    
+    DependencyNode node = resolveNode(groupId, artifactId, version);
+    
+    for (DependencyNode dependency : node.getChildren()) {
+      System.out.println(dependency.toString());
+    }
+  }
+
+  private static DependencyNode resolveNode(String groupId, String artifactId, String version)
+      throws DependencyCollectionException, DependencyResolutionException {
+    
+    RepositorySystemSession session = newSession();
+
+    Artifact artifact = new DefaultArtifact(groupId + ':' + artifactId + ':' + version);
+    Dependency dependency = new Dependency(artifact, "compile");
+
+    CollectRequest collectRequest = new CollectRequest();
+    collectRequest.setRoot(dependency);
+    collectRequest.addRepository(CENTRAL);
+    DependencyNode node = SYSTEM.collectDependencies(session, collectRequest).getRoot();
+
+    DependencyRequest dependencyRequest = new DependencyRequest();
+    dependencyRequest.setRoot(node);
+
+    SYSTEM.resolveDependencies(session, dependencyRequest);
+    return node;
+  }
+
+  private static RepositorySystemSession newSession() {
+    DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+
+    LocalRepository localRepo = new LocalRepository("target/local-repo");
+    session.setLocalRepositoryManager(SYSTEM.newLocalRepositoryManager(session, localRepo));
+    session.setReadOnly();
+    return session;
+  }
+
+  private static RepositorySystem newRepositorySystem() {
+    DefaultServiceLocator locator = MavenRepositorySystemUtils.newServiceLocator();
+    locator.addService(RepositoryConnectorFactory.class, BasicRepositoryConnectorFactory.class);
+    locator.addService(TransporterFactory.class, FileTransporterFactory.class);
+    locator.addService(TransporterFactory.class, HttpTransporterFactory.class);
+
+    return locator.getService(RepositorySystem.class);
+  }
+
+  /**
+   * Returns the direct compile time dependencies of an artifact.
+   */
+  public static List<Artifact> getImmediateDependencies(String groupId, String artifactId, String version) 
```

Comment:
Would it make sense to relocate this to be near `resolveNode()`?

Reply:
Seems odd to have "api" methods at the end.

Reply:
ack